src/ModelContextProtocol/Protocol/Types/Capabilities.cs

```diff
@@ -34,7 +34,7 @@ public class ClientCapabilities
     /// The client will not re-enumerate the sequence.
     /// </remarks>
     [JsonIgnore]
-    public IEnumerable<KeyValuePair<string, Func<JsonRpcNotification, Task>>>? NotificationHandlers { get; set; }
+    public IEnumerable<KeyValuePair<string, Func<JsonRpcNotification, CancellationToken, Task>>>? NotificationHandlers { get; set; }
```

Comment:
❤️ I just asked for this in my PR [here](https://github.com/modelcontextprotocol/csharp-sdk/pull/225/files#diff-abbb948b11cc86bef6ca3408a3c69b586abf4e826b0052c676a5a6e4fea1f29cR87-R89). I guess this means we'll definitely have merge conflicts.